### PR TITLE
Take BoM into account when searching

### PR DESCRIPTION
**Pull request by AbdulrhmnGhanem**

### CODE REVIEW

File: .idea/dictionaries/prodigy.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<component name="ProjectDictionaryState">
+  <dictionary name="prodigy">
+    <words>
+      <w>kitspace</w>
+    </words>
+  </dictionary>
+</component>
```
Comment (kasbah) [inline .idea/dictionaries/prodigy.xml:7]: Do we need to add all these `.idea` files? :worried:

File: package.json [hunk 2]
```diff
@@ -45,7 +45,7 @@
     "serve": "http-server build/",
     "lint": "eslint src",
     "fmt": "prettier --write configure *.js $(find src -name '*.jsx') $(find src -name '*.js') $(find tasks -name '*.js')",
-    "build": "./configure && ninja",
+    "build": "./configure && ninja -j 4",
```
Comment (kasbah) [inline package.json:48]: Why are you adding this? Ninja figures out how many processes to start based on your CPU count.

File: .idea/dictionaries/prodigy.xml (hunk 1 above)
Comment (AbdulrhmnGhanem) [reply .idea/dictionaries/prodigy.xml:7]: ok, just will gitignore them.

**Comment by kasbah** [comment]
Looks like it's not travis or my fault and the production build is still failing. You can try and run the production build locally by doing:

```
export PATH=$(pwd)/node_modules/bin:$PATH
./configure production
ninja clean && ninja
```

File: package.json (hunk 2 above)
Comment (AbdulrhmnGhanem) [reply package.json:48]: My machine froze multiple times building it! and I had to restart it.

Comment (kasbah) [reply package.json:48]: Oh, I see.

**Comment by ian-ross** [comment]
@kasbah It might be worth saying that if you want to replicate the production build, you need to get all the production boards. For the stuff I've been doing, I found some bugs in my code by building all the production boards locally -- there were some special cases I'd not considered that I discovered that way.

I think that means running the `plug_versions` script (`yarn plug-reg`), then running `get_boards` (`yarn get-boards`). You end up with a lot of stuff and it takes a long time to run, but it's probably easier to debug that way than on Travis.

**Comment by AbdulrhmnGhanem** [comment]
The build for the latest commit was not triggered on the branch. On my fork https://travis-ci.org/github/AbdulrhmnGhanem/kitspace, the build passed the previous failure point, the failure was related to Netlify(site not found).

**Comment by kasbah** [comment]
Did you see I invited into the Github org? https://github.com/orgs/kitspace/invitations/ 

Push to this branch please and see how the build does there: https://github.com/kitspace/kitspace/tree/use-bom-for-search

**Comment by AbdulrhmnGhanem** [comment]
Now it passes on the [use-bom-for-search](https://github.com/kitspace/kitspace/tree/use-bom-for-search) branch.

**Comment by kasbah** [comment]
Nice! That means it's live on http://use-bom-for-search.preview.kitspace.org/ I was able to search for "NE555" and see: "huh, these project use 555 timers", very cool!

Let's close this PR and open up a new one for that branch (add some links though so we can refer back).

**Pull request closed without merging**